tests/test_model.py

```diff
@@ -708,6 +863,9 @@ def test_recursive_edition_equivalence(self):
         work = Work()
         work.license_pools.extend([gutenberg2_pool])
 
+        # This is necessary for the work to have an id in the db.
+        self._db.flush()
```

Comment:
If the ID is the issue, I think it's because the ID is an auto-increment and Postgresql is in charge of it.

Take a look at create() in model.py. That calls db.add(new_object) and then db.flush(). So there's no shame in doing this, but you might want to use create(Work) to remove a bit of code.